nx_neptune/instance_management.py

```diff
@@ -290,6 +295,78 @@ def create_na_instance(config: Optional[dict] = None):
         )
 
 
+def create_na_instance_with_s3_import(s3_arn: str, config: Optional[dict] = None):
+    """Creates a new Neptune Analytics graph instance and imports data from S3.
+
+    This function creates a new Neptune Analytics graph instance and immediately starts
+    importing data from the specified S3 location. It handles the complete workflow:
+    1. Validates required permissions
+    2. Creates a new graph instance
+    3. Starts the import task
+    4. Returns a Future that can be awaited for completion
+
+    Args:
+        s3_arn (str): The S3 location containing CSV data (e.g., 's3://bucket-name/prefix/')
+        config (Optional[dict]): Optional dictionary of custom configuration parameters
+            to use when creating the Neptune Analytics instance. If not provided,
+            default settings will be applied.
+            All options listed under boto3 documentations are supported.
+
+            Reference:
+            https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/neptune-graph/client/create_graph_using_import_task.html
+
+    Returns:
+        asyncio.Future: A Future that resolves when the import completes and instance is available for computation work.
+
+    Raises:
+        Exception: If the Neptune Analytics instance creation or import task fails
+        ValueError: If the role lacks required permissions
+    """
+
+    # Permissions check
+    user_arn = boto3.client(SERVICE_STS).get_caller_identity()["Arn"]
+    iam_client = IamClient(role_arn=user_arn, client=boto3.client(SERVICE_IAM))
```

Comment:
make `client` here (defaults to boto3.client(SERVICE_IAM)) an optional input argument to this function. This will be helpful with the SessionManager.